pkg/verifier/aws/aws_verifier.go

```diff
@@ -0,0 +1,319 @@
+		// Unable to tag the instance
+		return "", handledErrors.NewGenericError(err)
+	}
+
+	return instanceID, nil
+}
+
+func (a *AwsVerifier) findUnreachableEndpoints(ctx context.Context, instanceID string) error {
+	var (
+		b64ConsoleLogs string
+		consoleLogs    string
+	)
+	// Compile the regular expressions once
+	reUserDataComplete := regexp.MustCompile(userdataEndVerifier)
+	reUnreachableErrors := regexp.MustCompile(`Unable to reach (\S+)`)
+	reGenericFailure := regexp.MustCompile(`(?m)^(.*Cannot.*)|(.*Could not.*)|(.*Failed.*)|(.*command not found.*)`)
+	reDockerFailure := regexp.MustCompile(`(?m)(docker)`)
+
+	input := &ec2.GetConsoleOutputInput{
+		InstanceId: awsTools.String(instanceID),
+		Latest:     awsTools.Bool(true),
+	}
+
+	a.writeDebugLogs("Scraping console output and waiting for user data script to complete...")
+
+	// Periodically scrape console output and analyze the logs for any errors or a successful completion
+	err := helpers.PollImmediate(30*time.Second, 4*time.Minute, func() (bool, error) {
+		consoleOutput, err := a.AwsClient.GetConsoleOutput(ctx, input)
+		if err != nil {
+			return false, handledErrors.NewGenericError(err)
+		}
+
+		if consoleOutput.Output != nil {
+			// In the early stages, an ec2 instance may be running but the console is not populated with any data
+			if len(*consoleOutput.Output) == 0 {
+				a.writeDebugLogs("EC2 console consoleOutput not yet populated with data, continuing to wait...")
+				return false, nil
+			}
+
+			// Store base64-encoded output for debug logs
+			b64ConsoleLogs = *consoleOutput.Output
+
+			// The console consoleOutput starts out base64 encoded
+			scriptOutput, err := base64.StdEncoding.DecodeString(*consoleOutput.Output)
+			if err != nil {
+				a.writeDebugLogs(fmt.Sprintf("Error decoding console consoleOutput, will retry on next check interval: %s", err))
+				return false, nil
+			}
+
+			consoleLogs = string(scriptOutput)
+
+			// Check for the specific string we consoleOutput in the generated userdata file at the end to verify the userdata script has run
+			// It is possible we get EC2 console consoleOutput, but the userdata script has not yet completed.
+			userDataComplete := reUserDataComplete.FindString(consoleLogs)
+			if len(userDataComplete) < 1 {
+				a.writeDebugLogs("EC2 console consoleOutput contains data, but end of userdata script not seen, continuing to wait...")
+				return false, nil
+			}
+
+			// Check consoleOutput for failures, report as exceptions if they occurred
+			genericFailures := reGenericFailure.FindAllStringSubmatch(consoleLogs, -1)
+			if len(genericFailures) > 0 {
+				a.writeDebugLogs(fmt.Sprint(genericFailures))
+
+				dockerFailures := reDockerFailure.FindAllString(consoleLogs, -1)
+				if len(dockerFailures) > 0 {
+					// Should be resolved by OSD-13003 and OSD-13007
+					a.Output.AddException(handledErrors.NewGenericError(errors.New("docker was unable to install or run. Further investigation needed")))
```

Comment:
I got this when executing against a subnet that has internet connection (public  - there's route table to igw)
```
./osd-network-verifier egress --subnet-id subnet-0ed916fb8a22b3342 --profile admintestaccount --region us-east-1
Using region: us-east-1
Created instance with ID: i-0580a591d84ba50a5
Summary:
printing out exceptions preventing the verifier from running the specific test:
 -  network verifier error: docker was unable to install or run. Further investigation needed
printing out errors faced during the execution:
 -  network verifier error: network verifier error: [docker docker docker docker docker docker docker]
Failure!
```

Reply:
+1 same error for private subnets too

Reply:
I ended up fixing this in [e528c88](https://github.com/openshift/osd-network-verifier/pull/122/commits/e528c887c4fd24e02b7cf01f5cd43d87e3429af9)

Reply:
We fixed the issue by fixing the USER DATA variables as well as ensuring the ec2 instnace has a public IP